core_java/jdbc/src/main/java/ca/jrvs/apps/jdbc/dao/OrderDAO.java

```diff
@@ -0,0 +1,134 @@
+                orderLine.setProductName(resultSet.getString(13));
+                orderLine.setProductSize(resultSet.getInt(14));
+                orderLine.setProductVariety(resultSet.getString(15));
+                orderLine.setProductPrice(resultSet.getBigDecimal(16));
+                orderLines.add(orderLine);
+            }
+            order.setOrderLines(orderLines);
+        }catch(SQLException e){
+            logger.error("ERROR: ORDER FIND BY ID");
+            throw new RuntimeException(e);
+        }
+        return order;
+    }
+
+    @Override
+    public List<Order> findAll() {
+        return null;
+    }
+
+    @Override
+    public Order update(Order dto) {
+        return null;
+    }
+
+    @Override
+    public Order create(Order dto) {
+        return null;
+    }
+
+    @Override
+    public void delete(long id) {
+
+    }
+
+    public List<Order> getOrdersForCustomer(long customerId){
+        List<Order> orders = new ArrayList<>();
+        try(PreparedStatement statement = this.connection.prepareStatement(GET_FOR_CUST);){
+            statement.setLong(1, customerId);
+            ResultSet resultSet = statement.executeQuery();
+            long orderId = 0;
+            Order order = null;
+            while(resultSet.next()){
+                long localOrderId = resultSet.getLong(4);
+                if(orderId!=localOrderId){
+                    order = new Order();
+                    orders.add(order);
+                    order.setId(localOrderId);
+                    orderId = localOrderId;
+                    order.setCustomerFirstName(resultSet.getString(1));
+                    order.setCustomerLastLane(resultSet.getString(2));
+                    order.setCustomerEmail(resultSet.getString(3));
+                    order.setCreationDate(new Date(resultSet.getDate(5).getTime()));
+                    order.setTotalDue(resultSet.getBigDecimal(6));
+                    order.setStatus(resultSet.getString(7));
+                    order.setSalespersonFirstName(resultSet.getString(8));
+                    order.setSalespersonLastName(resultSet.getString(9));
+                    order.setSalespersonEmail(resultSet.getString(10));
+                    List<OrderLine> orderLines = new ArrayList<>();
+                    order.setOrderLines(orderLines);
+                }
+                OrderLine orderLine = new OrderLine();
+                orderLine.setQuantity(resultSet.getInt(11));
+                orderLine.setProductCode(resultSet.getString(12));
+                orderLine.setProductName(resultSet.getString(13));
+                orderLine.setProductSize(resultSet.getInt(14));
+                orderLine.setProductVariety(resultSet.getString(15));
+                orderLine.setProductPrice(resultSet.getBigDecimal(16));
+                order.getOrderLines().add(orderLine);
+            }
+        }catch(SQLException e){
+            logger.error("ERROR: GET ORDER FOR CUSTOMER");
```

Comment:
don't log message befoe thrwoing a new exception as you will double log the same message

           ` throw new RuntimeException("GET ORDER FOR CUSTOMER", e);`